src/linalg/eigen.ts

```diff
@@ -0,0 +1,123 @@
+import { linalg, Tensor, matMul, abs, sub, max, tensor, mul, eye, slice, stack, squeeze, neg, transpose } from '@tensorflow/tfjs-core';
+import { linSolveQR } from './linsolve';
+import { tensorNormalize, tensorEqual } from './utils';
+/**
+ * Compute the eigenvalues of a matrix using the QR algorithm.
+  This is a renormalized version of power iteration that converges to a full
+  set of eigenvalues.  Starting with the matrix M = M0, we iterate:
+    M0 = Q0 R0, M1 = R0 Q0;
+    M1 = Q1 R1, M2 = R1 Q1;
+    M2 = Q2 R2, M3 = R2 Q2;
+    ...
+  For a general matrix with a full set of eigenvalues, this sequence will
+  converge to an upper diagonal matrix:
+    Mi -> upper diagonal matrix
+  The diagonal entries of this matrix are the eigenvalues of M.
+  Qn is the eigenvector of M.
+ * @param matrix square matrix of shape (n, n)
+   @param tol tolerence, default to 1e-4
+   @param maxIter max iteration time, default to 200
+ */
+export const solveEigenValues = (matrix: Tensor, tol = 1e-4, maxIter = 200): Tensor => {
+  let [ q, r ] = linalg.qr(matrix);
+  let x = matrix;
+  let prevX: Tensor;
+  let prevTr: Tensor;
+  const n = matrix.shape[0];
+  let xTr = linalg.bandPart(x, 0, 0);
+  let qn = q;
+
+  for (let i = 0; i < maxIter; i++) {
+    prevX = x;
+    x = matMul(r, q);
+    [ q, r ] = linalg.qr(x);
+    qn = matMul(qn, q);
+    xTr = linalg.bandPart(x, 0, 0);
+    prevTr = linalg.bandPart(prevX, 0, 0);
+    const maxDis = max(abs(sub(prevTr, xTr))).arraySync();
+    if ( maxDis < tol) {
+      break;
+    }
+  }
+  x = matMul(r, q);
+  const eigenValues = [];
+  for (let i = 0; i < n; i++) {
+    eigenValues.push(slice(x, [ i, i ], [ 1, 1 ]));
+  }
+  const eigenValuesTensor = squeeze(stack(eigenValues));
```

Comment:
```suggestion
  return squeeze(stack(eigenValues));
```

No necessary to declaration & assignment ops.